src/platform/packages/shared/kbn-esql-ast/src/commands_registry/commands/rename/columns_after.ts

```diff
@@ -31,25 +30,35 @@ export const columnsAfter = (
   }
 
   // rename the columns with the user defined name
-  const newFields = previousColumns.map((oldColumn) => {
+  const newFields = previousColumns.map<ESQLColumnData>((oldColumn) => {
     const asRenamePair = asRenamePairs.find(
       (pair) => (pair?.args?.[0] as ESQLAstBaseItem)?.name === oldColumn.name
     );
 
-    if (asRenamePair?.args?.[1]) {
-      return { name: (asRenamePair.args[1] as ESQLAstBaseItem).name, type: oldColumn.type };
+    if (isColumn(asRenamePair?.args?.[1])) {
+      return {
+        name: asRenamePair.args[1].name,
+        type: oldColumn.type,
+        location: asRenamePair.args[1].location,
+        userDefined: true,
+      };
     }
 
     const assignRenamePair = assignRenamePairs.find(
       (pair) => (pair?.args?.[1] as ESQLAstBaseItem)?.name === oldColumn.name
     );
 
-    if (assignRenamePair?.args?.[0]) {
-      return { name: (assignRenamePair.args[0] as ESQLAstBaseItem).name, type: oldColumn.type };
+    if (isColumn(assignRenamePair?.args?.[0])) {
+      return {
+        name: assignRenamePair.args[0].name,
+        type: oldColumn.type,
+        location: assignRenamePair.args[0].location,
+        userDefined: true,
+      };
     }
 
     return oldColumn; // No rename found, keep the old name
   });
 
-  return uniqBy(newFields, 'name');
+  return uniqBy(newFields, 'name') as ESQLColumnData[];
```

Comment:
Is this needed?

Reply:
Nope https://github.com/elastic/kibana/pull/233221/commits/a6857e6a7f6abb73a76c47b2b4bfec62794bd31a